### PR TITLE
`Development`: Update client documentation to use standalone components

### PR DESCRIPTION
**Pull request by JohannesWt**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

As discussed today in the dev meeting we will use standalone components from now on for new components. Old components should be migrated step by step (evaluated case by case)

### Description

This PR updates the client documentation to use standalone components.
Changed documentation https://artemis-platform--8678.org.readthedocs.build/en/8678/dev/guidelines/client.html#components

#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2

### CODE REVIEW

File: docs/dev/guidelines/client.rst [hunk 1]
```diff
@@ -35,6 +35,12 @@ Some general aspects:
 2. Components
 =============
 
+In our project, we advocate for the creation of standalone components. A standalone component is a self-contained unit that encapsulates its own logic, view, and styles. It doesn't directly depend on its parent or child components and can be reused in different parts of the application.
```
Comment (pzdr7) [inline docs/dev/guidelines/client.rst:38]: Could we explicitly mention modules as the alternative since it's currently what we have in Artemis?

**Review by JohannesStoehr — APPROVED** [review]
New documentation looks good to me

**Review by pzdr7 — APPROVED** [review]
LGTM

**Review by rabeatwork — APPROVED** [review]
maintainer approve

**Review by muradium — APPROVED** [review]
LGTM

**Review by rstief — APPROVED** [review]

**Pull request merged**